src/notification/index.js

```diff
@@ -25,15 +25,15 @@ import { getAccounts } from '../directSelectors';
 /**
  * Identify the account the notification is for, if possible.
  *
- * Returns an index into `identities`, or `null` if we can't tell.
+ * Returns an index into `accounts`, or `null` if we can't tell.
  *
- * @param identities Identities corresponding to the accounts state in Redux.
+ * @param accounts The accounts state in Redux.
  */
 export const getAccountFromNotificationData = (
   data: Notification,
-  identities: $ReadOnlyArray<Identity>,
+  accounts: $ReadOnlyArray<Account>,
```

Comment:
The Identity type doesn't have the user ID, and it's not super
    convenient to add it there. That's because the Identity type is
    based on the Auth type (and we make Identity objects out of Auth
    objects at runtime), and the Auth type doesn't have the user ID.
    It's not super convenient to give it to the Auth type because there
    are places where we make Auth objects but we don't have a user ID,
    like `authFromCallbackUrl` in src/start/webAuth.js.

Oof. The `Auth` does of course have the email. So anywhere we have an Auth and can't readily make an Identity, because we lack the user ID, is an obstacle to converting from emails to IDs, #3764.

In `authFromCallbackUrl`, we're decoding one of those `zulip://` URLs the server used to complete a web login and get us the API key. So we should get the server to start giving us user IDs there, even if we won't be able to count on their presence for a while. That should be a thread in `#api design`, then.